core/controlplane/config/templates/cloud-config-worker

```diff
@@ -391,7 +391,7 @@ coreos:
             } \
           } \
           }\"' \
-          https://{{.ExternalDNSName}}:443/api/v1/nodes/$(hostname)"
+          https://{{.APIEndpoint.DNSName}}:443/api/v1/nodes/$(hostname)"
```

Comment:
Perhaps we'd better provide something like:

```golang
func (c ProvidedConfig) ExternalDNSName() string {
    fmt.Println("WARN: ExternalDNSName is deprecated and will be removed in v0.9.7. Please use APIEndpoint.Name instead")
    return c.APIEndpint.DNSName
}
```

so that customized `cloud-config-worker` files or changes from other PRs still referencing `.ExternalDNSName` won't break.